MapboxDirectionsTests/MatchTests.swift

```diff
@@ -0,0 +1,106 @@
+    }
+    
+    func test() {
+        let expectation = self.expectation(description: "calculating directions should return results")
+        let locations = [CLLocation(coordinate: CLLocationCoordinate2D(latitude: 37.78, longitude: -122.42)),
+                         CLLocation(coordinate: CLLocationCoordinate2D(latitude: 38.91, longitude: -77.03))]
+        
+        let queryParams: [String: String?] = [
+            "geometries": "polyline",
+            "overview": "full",
+            "steps": "true",
+            "tidy": "false",
+            "access_token": BogusToken,
+            ]
+        stub(condition: isHost("api.mapbox.com")
+            && isPath("/matching/v5/mapbox/driving/-122.42,37.78;-77.03,38.91.json")
+            && containsQueryParams(queryParams)) { _ in
+                let path = Bundle(for: type(of: self)).path(forResource: "match", ofType: "json")
+                return OHHTTPStubsResponse(fileAtPath: path!, statusCode: 200, headers: ["Content-Type": "application/json"])
+        }
+        
+        var match: Match!
+        var tracePoints: [Tracepoint]!
+        let matchOptions = MatchingOptions(locations: locations)
+        matchOptions.includesSteps = true
+        matchOptions.routeShapeResolution = .full
+        
+        let task = Directions(accessToken: BogusToken).match(matchOptions) { (tPoints, matches, error) in
+            XCTAssertNil(error, "Error: \(error!)")
+            
+            match = matches!.first!
+            tracePoints = tPoints
+            
+            expectation.fulfill()
+        }
+        XCTAssertNotNil(task)
+        
+        waitForExpectations(timeout: 2) { (error) in
+            XCTAssertNil(error, "Error: \(error!)")
+            XCTAssertEqual(task.state, .completed)
+        }
+        
+        XCTAssertNotNil(match)
+        XCTAssertNotNil(match.coordinates)
+        XCTAssertEqual(match.coordinates!.count, 8)
+        XCTAssertEqual(match.accessToken, BogusToken)
+        XCTAssertEqual(match.apiEndpoint, URL(string: "https://api.mapbox.com"))
+        XCTAssertEqual(match.routeIdentifier, nil)
+        
+        XCTAssertNotNil(tracePoints)
+        XCTAssertEqual(tracePoints.first!.alternateCount, 0)
+        XCTAssertEqual(tracePoints.first!.matchingIndex, 0)
+        XCTAssertEqual(tracePoints.first!.waypointIndex, 0)
+        
+        XCTAssertEqual(tracePoints.last!.name, "West G Street")
+        XCTAssertEqual(tracePoints.last!.waypointIndex, 6)
+
+        // confirming actual decoded values is important because the Directions API
+        // uses an atypical precision level for polyline encoding
+        XCTAssertEqual(round(match!.coordinates!.first!.latitude), 33)
+        XCTAssertEqual(round(match!.coordinates!.first!.longitude), -117)
+        XCTAssertEqual(match!.legs.count, 6)
+        XCTAssertEqual(match!.confidence, 0.95)
+
+        let opts = match!.directionOptions
+
+        XCTAssertEqual(opts, matchOptions)
```

Comment:
This assertion is failing and might point out a bigger issue with this approach. Is is possible to override a superclasses type? If so, in `Match`, we should override `directionOptions` to be of type `MatchOptions`. Perhaps the issue here is that we're casting the type.

Reply:
> If so, in Match, we should override directionOptions to be of type MatchOptions.

Yes, that’s called a covariant property.

Reply:
Since a covariant property isn’t possible here, add a `matchingOptions` property to Match that returns a casted `directionsOptions`. Then assert here that it’s equal to the local `matchOptions` variable.